File: mqtt/src/main/java/com/aleksa/iots/iots/model/SensorReading.java
package com.aleksa.iots.iots.model;

import jakarta.persistence.*;
import java.time.Instant;

@Entity
@Table(name = "sensor_readings")
public class SensorReading {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Instant timestamp;

    @Column(name = "device_id")
    private String deviceId;

    private Double temperature;
    private Double humidity;
    private Double pressure;
    private Integer light;
    private Integer sound;
    private Short motion;
    private Double battery;
    private String location;

    private String broker;

    @Column(name = "received_at")
    private Instant receivedAt;

    @Column(name = "sent_at")
    private Instant sentAt;

    @PrePersist
    public void prePersist() {
        this.receivedAt = Instant.now();
        this.broker = "mqtt";
    }

    public Long getId() { return id; }
    public Instant getTimestamp() { return timestamp; }
    public void setTimestamp(Instant timestamp) { this.timestamp = timestamp; }
    public String getDeviceId() { return deviceId; }
    public void setDeviceId(String deviceId) { this.deviceId = deviceId; }
    public Double getTemperature() { return temperature; }
    public void setTemperature(Double temperature) { this.temperature = temperature; }
    public Double getHumidity() { return humidity; }
    public void setHumidity(Double humidity) { this.humidity = humidity; }
    public Double getPressure() { return pressure; }
    public void setPressure(Double pressure) { this.pressure = pressure; }
    public Integer getLight() { return light; }
    public void setLight(Integer light) { this.light = light; }
    public Integer getSound() { return sound; }
    public void setSound(Integer sound) { this.sound = sound; }
    public Short getMotion() { return motion; }
    public void setMotion(Short motion) { this.motion = motion; }
    public Double getBattery() { return battery; }
    public void setBattery(Double battery) { this.battery = battery; }
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    public String getBroker() { return broker; }
    public Instant getReceivedAt() { return receivedAt; }
    public Instant getSentAt() { return sentAt; }
    public void setSentAt(Instant sentAt) { this.sentAt = sentAt; }
}

File: kafka/storage-service/SensorReading.cs
using System.Text.Json.Serialization;

namespace StorageService;

public class SensorReading
{
    [JsonPropertyName("timestamp")]
    public DateTime Timestamp { get; set; }

    [JsonPropertyName("device_id")]
    public string DeviceId { get; set; } = "";

    [JsonPropertyName("temperature")]
    public double Temperature { get; set; }

    [JsonPropertyName("humidity")]
    public double Humidity { get; set; }

    [JsonPropertyName("pressure")]
    public double Pressure { get; set; }

    [JsonPropertyName("light")]
    public int Light { get; set; }

    [JsonPropertyName("sound")]
    public int Sound { get; set; }

    [JsonPropertyName("motion")]
    public int Motion { get; set; }

    [JsonPropertyName("battery")]
    public double Battery { get; set; }

    [JsonPropertyName("location")]
    public string Location { get; set; } = "";

    [JsonPropertyName("sent_at")]
    public DateTime SentAt { get; set; }
}
